### PR TITLE
docs: use-private-ip annotation

### PR DESCRIPTION
**Pull request by akshaysngupta**
This PR adds documentation for use-private-ip annotation.

### CODE REVIEW

File: docs/annotations.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Kubernetes Ingress specification allows for annotations. We use annotations to e
 | [appgw.ingress.kubernetes.io/connection-draining-timeout](#connection-draining) | `int32` (seconds) | `30` |
 | [appgw.ingress.kubernetes.io/cookie-based-affinity](#cookie-based-affinity) | `bool` | `false` |
 | [appgw.ingress.kubernetes.io/request-timeout](#request-timeout) | `int32` (seconds) | `30` |
+| [appgw.ingress.kubernetes.io/use-private-ip](#use-private-ip) | `int32` (seconds) | `30` |
```
Comment (delqn) [inline docs/annotations.md]: The `int32 seconds` does not sound right :)
Should this be 
```
 | `bool` | `false` |
```
?

File: docs/annotations.md [hunk 2]
```diff
@@ -183,3 +184,37 @@ spec:
           serviceName: go-server-service
           servicePort: 80
 ```
+
+## Use Private IP
+
+This annotation allows to specify whether to expose this endpoint on Private IP of Application Gateway.
```
Comment (delqn) [inline docs/annotations.md]: This sentence is missing a pronoun it seems. Should this be `This annotation allows us to specify`. Aadding `us`, or `you` will help.

File: docs/annotations.md [hunk 3]
```diff
@@ -183,3 +184,37 @@ spec:
...
+
+> **Note**
+1) Application Gateway doesn't support multiple IPs on the same port (example: 80/443). So, if you have one ingress with annotation `use-private-ip: "false"` and other with `use-private-ip: "true"` on `HTTP`, they will fail the update on the Gateway.
```
Comment (delqn) [inline docs/annotations.md]: I'd remove `So, if you have one ` and leave it at `Ingress with annotation`
also I'd change `other` with `another`.
I'd change `they will` with `AGIC will fail to update...`

File: docs/annotations.md [hunk 4]
```diff
@@ -183,3 +184,37 @@ spec:
...
+2) In case where Application Gateway doesn't have a private IP, specific ingresses with `use-private-ip: "true"` be pruned out. This will reflected in the controller logs and ingress events for those ingresses with `NoPrivateIP` .
```
Comment (delqn) [inline docs/annotations.md]: I wonder if we should rephrase this as `App Gateways that don't have a private IP and are annotated with … will be ignored`.

File: docs/features/private-ip.md [hunk 5]
```diff
@@ -1,30 +1,53 @@
 # Using Private IP for internal routing
 
-## Pre-requisites
-* Application Gateway with a [Private IP configuration](https://docs.microsoft.com/en-us/azure/application-gateway/configure-application-gateway-with-private-frontend-ip)
+This feature allows to expose the ingress endpoint within the `Virtual Network` using a private IP.
 
-This feature allows to expose the ingress endpoint within the `Virtual Network`.
-To configure the controller to use Private IP for routing, modify the `helm` config by adding `usePrivateIP: true`
+**Pre-requisites**
```
Comment (delqn) [inline docs/features/private-ip.md]: GitHub Markdown renders this on the same line as the next. You could add another `\n` or keep the `##` and add another `#`?

File: docs/features/private-ip.md [hunk 6]
```diff
@@ -1,30 +1,53 @@
...
+
+## Assign to a particular ingress
+To expose a particular ingress over Private IP, use annotation [`use-private-ip`](../annotations.md#use-private-ip) in Ingress.
+
+### Usage
+```yaml
+appgw.ingress.kubernetes.io/use-private-ip: "true"
+```
+
+If Application Gateway doesn't have a private IP, then Ingresses with this annotation set to `true` will be pruned. This will be indicated in the ingress event and AGIC pod log.
```
Comment (delqn) [inline docs/features/private-ip.md]: I'd recommend we simplify this sentence to - `App Gateways without a Private IP and annotated with … will be ignored`

File: docs/features/private-ip.md [hunk 7]
```diff
@@ -1,30 +1,53 @@
...
 
 This will make the ingress controller filter the ipconfigurations for a Private IP when configuring the frontend listeners on the Application Gateway.
-Controller will panic and crash if `usePrivateIP: true` and no Private IP is assigned.
+AGIC will panic and crash if `usePrivateIP: true` and no Private IP is assigned.
 
 **Notes:**
-
-1. Application Gateway v2 SKU manadates a Public IP. For meeting compliance requirement where the Application Gateway should be completely private, Attach a [`Network Security Group`](https://docs.microsoft.com/en-us/azure/virtual-network/security-overview) to the Application Gateway's subnet to restrict traffic.
-1. To expose the Ingress both on public and private, current recommedation is to deploy two Application Gateways with respective Ingress Controllers, one with Public IP and one with Private IP.
+Application Gateway v2 SKU manadates a Public IP. For meeting compliance requirement where the Application Gateway should be completely private, Attach a [`Network Security Group`](https://docs.microsoft.com/en-us/azure/virtual-network/security-overview) to the Application Gateway's subnet to restrict traffic.
```
Comment (delqn) [inline docs/features/private-ip.md]: s/mandates/requires/ ?

Comment (delqn) [inline docs/features/private-ip.md]: `compliance` could be one of the reasons.  I'd rephrase this as `Should you require Application Gateway to be private...`

File: docs/annotations.md (hunk 1 above)
Comment (akshaysngupta) [reply docs/annotations.md]: ops!

**Comment by akshaysngupta** [comment]
Thanks @delqn. PTAL
